crates/wash-cli/src/cmd/up.rs

```diff
@@ -914,6 +929,40 @@ pub(crate) async fn start_nats(
     Ok(nats_process)
 }
 
+/// Helper function to run an optimistic patch update, but fall back to the previous version if the new version fails
+/// to download.
+async fn install_patch_or_default_wadm_version(
+    version: &Option<String>,
+    install_dir: &Path,
+) -> Result<PathBuf> {
+    if let Some(version) = version {
+        return ensure_wadm(version, install_dir).await;
+    }
+    const WASMCLOUD_ORG: &str = "wasmCloud";
+    const WASMCLOUD_WADM_REPO: &str = "wadm";
+    let version = version.clone().unwrap_or(WADM_VERSION.to_owned());
+    let new_patch_version =
+        new_patch_version_of_after_string(WASMCLOUD_ORG, WASMCLOUD_WADM_REPO, &version).await?;
+    match new_patch_version {
+        Some(new_patch) => {
+            let new_version = format!("v{}", new_patch);
```

Comment:
Is the new patch version returned by `new_patch_version`... always the semver format without the `v`?

Reply:
Absolutely good question there are places where the `v` is needed, I probably will try to centralize the different version of these on the GitHubAction struct or with its Semver field. It is slightly confusing to use, so would make sense to make it more helpful for others if they will use it.

Reply:
Yes, the semver create is flexible when parsing the version, it is treating the "v" prefix as option, but nor the display cannot differentiate the original value.
https://docs.rs/semver/1.0.23/src/semver/display.rs.html#93
If we would (will as I've seen wit the nats) use it in multiple cases we can add a method to the `GitHubRelease` itself. But don't want to prematurely add options that I do not know.